src/engine/GraphStoreProtocol.h

```diff
@@ -44,18 +32,23 @@ class GraphStoreProtocol {
       // If the mediatype is not given, return an error.
       // Note: The specs also allow to try to determine the media type from the
       // content.
-      throw UnknownMediatypeError("Mediatype empty or not set.");
+      throw HttpError(boost::beast::http::status::unsupported_media_type,
+                      "Mediatype empty or not set.");
+    }
+    std::optional<std::vector<ad_utility::MediaType>> mediaTypes;
+    try {
+      mediaTypes = ad_utility::getMediaTypesFromAcceptHeader(contentTypeString);
+    } catch (const std::exception& e) {
+      throw HttpError(boost::beast::http::status::unsupported_media_type,
+                      e.what());
```

Comment:
1. Belongs in the HttpError PR and
2.  The interface has changed, make sure that the `getMediatTypes... never throws and then you don't need to catch here, but just throw.